test-data/unit/check-dataclasses.test

```diff
@@ -390,9 +389,10 @@ app1 = Application("example-1", 5)
 app2 = Application("example-2", 5)
 app1 == app2
 app1 != app2
-app1 == None  # E: Unsupported operand types for == ("Application" and "None")
+app1 == None
+reveal_type(app1.__eq__)  # N: Revealed type is 'def (builtins.object) -> builtins.bool'
```

Comment:
@ilevkivskyi maybe this test is not valid. Possible it passes on master because of incomplete `object` definition in fixtures/list.pyi stub, in dataclasses.pyi `object` has `__eq__` and `__ne__` methods.

Example:
```python
from dataclasses import dataclass

@dataclass
class A:
    pass

reveal_type(A.__eq__)
```
On latest version of mypy output is:
```
test.py:7: note: Revealed type is 'def (builtins.object, builtins.object) -> builtins.bool'
```
Maybe this code doesn't work: https://github.com/python/mypy/blob/master/mypy/plugins/dataclasses.py#L127
And I don't understand, why `__eq__` has this type: `(self: T, other: T) -> bool`. Maybe need to create new issue for this?